### PR TITLE
Added encrypted_value to Actions + Organizations's secrets

### PR DESCRIPTION
**Pull request by nadenf**
Fixes:

#704 - Use encrypted_value in github_actions_organization_secret and github_actions_secret

### CODE REVIEW

**Review by tibbes — CHANGES_REQUESTED** [review]
Thanks for your pull requests @nadenf! I think @jcudit is going to post a fuller review shortly, but I just wanted to leave a quick note on the blocker for merging this.

File: github/data_source_github_repository_branches.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package github
```
[the remark this answers is not in the document]
Comment (jcudit) [reply github/data_source_github_repository_branches.go]: nvm, removed in 6880b90

**Comment by nadenf** [comment]
@jcudit @tibbes  .. All changes made.

**Review by tibbes — APPROVED** [review]
Thanks @nadenf. 👍 on the base64 changes. I've also manually tested that using `encrypted_value` works (for repo-level secrets).

**Comment by tibbes** [comment]
But we seem to have lost 6880b90a3cb40a95fa5501c461c7968849e99e9a (the commit that removes `data_source_github_repository_branches.go`)

**Comment by nadenf** [comment]
@tibbes  .. File has been removed.

**Review by jcudit — APPROVED** [review]

**Comment by hoxu** [comment]
Is there documentation somewhere on how to create the `encrypted_value` using the public key?

**Comment by jcudit** [comment]
@hoxu a (hack) workaround would be to:
- configure your `_secret` resource using a `plaintext_value`
- `TF_LOG=DEBUG terraform apply ...` the configuration
- extract the encrypted and base64 encoded string that is sent to the GitHub API
- use the extracted value in the `encrypted_value` field going forward

I have not tried this personally but theoretically this should work if I am not overlooking something.

**Comment by hoxu** [comment]
@jcudit thanks, this indeed works. But if you have remote state (especially with versioning), you need to manually comment out the backend, `terraform init`, do all this, enable the backend again and `terraform init`. In my opinion it's too cumbersome. And these steps are not documented on https://registry.terraform.io/providers/integrations/github/latest/docs/resources/actions_secret

**Pull request merged**